tiledb/api/src/query/read/mod.rs

```diff
@@ -374,6 +384,262 @@ pub trait ReadQueryBuilder<'data>: QueryBuilder {
     }
 }
 
+pub struct AggregateBuilder<B, T> {
+    base: B,
+    agg_str: CString,
+    attr_str: Option<CString>,
+    attr_type: PhantomData<T>,
+}
+
+pub struct AggregateReader<Q, T> {
+    base: Q,
+    agg_str: CString,
+    _attr_str: Option<CString>,
+    data : T,
+    data_size : u64
+}
+
+impl<B, T> QueryBuilder for AggregateBuilder<B, T>
+where
+    B: QueryBuilder,
+    T : Default,
```

Comment:
When we want to say that a type is supposed to be something that comes out of a tiledb query, I think we're best off using `T: PhysicalType` from our `datatype` module.

Is that more restrictive than is really needed here?  Yes.  But I think it's a better expression of intent.

Reply:
fixed!